### PR TITLE
fix: cloning alert 

### PR DESCRIPTION
**Pull request by palashgdev**
close: https://github.com/SigNoz/signoz/issues/4034

after: 

https://github.com/SigNoz/signoz/assets/88981777/99965853-5369-4e81-ae3a-b64afc0438ef

before: 

https://www.loom.com/share/24ff63a055d04d4081d4f79430e817ad

### CODE REVIEW

**Review by srikanthccv — APPROVED** [review]

**Comment by YounixM** [comment]
@palashgdev, @srikanthccv : Are cloned alerts enabled by default?

**Comment by YounixM** [comment]
image

**Comment by YounixM** [comment]
image


@palashgdev: Changes don't seem to work as expected.

**Comment by palashgdev** [comment]
> image
> 
> @palashgdev: Changes don't seem to work as expected.

please share a loom video image does not convey anything

**Comment by YounixM** [comment]
Cloned a disabled alert , the status is Ok and in options we get enable option.

**Comment by palashgdev** [comment]
> @Rajat-Dabade: Cloned a disabled alert, the status is Ok and in options we get the enable option.

i have already shared the context to @srikanthccv he have to look into this one... though nothing wrong from client side... as there is status mismatch from backend itself

**Comment by YounixM** [comment]
@palashgdev : Not sure exactly what to verify in the PR then. The title, the before video tell the issue and this PR doesn't fix the mentioned issue.

**Comment by palashgdev** [comment]
@YounixM @srikanthccv since client issue is resolved can you merge this PR

**Comment by srikanthccv** [comment]
@palashgdev we will fix the remaining concerns and then get this merged. This is also not a critical bug.

**Comment by palashgdev** [comment]
> @palashgdev we will fix the remaining concerns and then get this merged. This is also not a critical bug.

@YounixM

File: frontend/src/container/ListAlertRules/index.tsx [hunk 1]
```diff
@@ -13,9 +13,10 @@ import ListAlert from './ListAlert';
 function ListAlertRules(): JSX.Element {
 	const { t } = useTranslation('common');
 	const location = useLocation();
-	const { data, isError, isLoading, refetch, status } = useQuery('allAlerts', {
+	const { data, isError, isLoading, status } = useQuery('allAlerts', {
 		queryFn: getAll,
 		cacheTime: 0,
+		refetchInterval: 30000,
```
Comment (YounixM) [inline frontend/src/container/ListAlertRules/index.tsx:19]: @palashgdev: Can we consider any other approach than polling? We can rely on the clone API response to refetch the alerts.

Comment (palashgdev) [reply frontend/src/container/ListAlertRules/index.tsx:19]: @YounixM earlier we were dependent over refetch method itself over not this is handled by react-query itself

**Comment by palashgdev** [comment]
> @palashgdev: When working on a component with code inside `index.ts`, let's move the code to the file named as the component name and reference it in index.ts as planned before.

sure this PR was is not taking care of this one making a ticket to track the changes

File: frontend/src/container/ListAlertRules/index.tsx (hunk 1 above)
Comment (YounixM) [reply frontend/src/container/ListAlertRules/index.tsx:19]: Sorry, didn't understand the comment.

Comment (palashgdev) [reply frontend/src/container/ListAlertRules/index.tsx:19]: there is useInternal in `https://github.com/SigNoz/signoz/pull/4082/files#diff-c3f4ae491202b106350fa9a32c645177e941926d48505e9f4647d83e9c1961b1` over that we are refetching the alerts... directly handling custom logic ... used inbuilt params for our use case

**Comment by srikanthccv** [comment]
I forgot the details. Let me recollect the issue and update you.

**Pull request closed without merging**